[BE][DDPOptimizer] De-dup `p` and `param`

awgu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #95654

The `param` from `param = target.get_parameter(name)` should be the same as `p` from `target.named_parameters()`.

cc @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

Commit:
[BE][DDPOptimizer] De-dup `p` and `param`

[ghstack-poisoned]

wconstab reviewed (approved):
lgtm! nit: 'param' is a nicer name to keep

Commit:
Update on "[BE][DDPOptimizer] De-dup `p` and `param`"


The `param` from `param = target.get_parameter(name)` should be the same as `p` from `target.named_parameters()`.

cc @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

[ghstack-poisoned]

awgu commented:
@pytorchbot rebase -s

Commit:
Update on "[BE][DDPOptimizer] De-dup `p` and `param`"


The `param` from `param = target.get_parameter(name)` should be the same as `p` from `target.named_parameters()`.

cc soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

awgu commented:
@pytorchbot merge
